test/widgets/content_test.dart

```diff
@@ -282,6 +282,102 @@ void main() {
+                child: MessageContent(
+                  message: message,
+                  content: parseContent(html)))))));
+      await tester.pump(); // global store
+      await tester.pump(); // per-account store
+      debugNetworkImageHttpClientProvider = null;
+    }
+
+    testWidgets('single image', (tester) async {
+      // "https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3"
+      await prepareContent(tester,
+        '<div class="message_inline_image">'
+          '<a href="https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3">'
+            '<img src="https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3"></a></div>');
+      tester.widget(find.byType(RealmContentNetworkImage));
+      final images = tester.widgetList<RealmContentNetworkImage>(find.byType(RealmContentNetworkImage));
+      check(images.map((i) => i.src.toString()).toList())
+        .deepEquals([
+          'https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3'
+        ]);
+    });
+
+    testWidgets('parse multiple images', (tester) async {
+      // "https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3\nhttps://chat.zulip.org/user_avatars/2/realm/icon.png?version=4"
+      await prepareContent(tester,
+        '<p>'
+          '<a href="https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3">https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3</a><br>\n'
+          '<a href="https://chat.zulip.org/user_avatars/2/realm/icon.png?version=4">https://chat.zulip.org/user_avatars/2/realm/icon.png?version=4</a></p>\n'
+        '<div class="message_inline_image">'
+          '<a href="https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3">'
+            '<img src="https://uploads.zulipusercontent.net/f535ba07f95b99a83aa48e44fd62bbb6c6cf6615/68747470733a2f2f636861742e7a756c69702e6f72672f757365725f617661746172732f322f7265616c6d2f69636f6e2e706e673f76657273696f6e3d33"></a></div>'
+        '<div class="message_inline_image">'
+          '<a href="https://chat.zulip.org/user_avatars/2/realm/icon.png?version=4">'
+            '<img src="https://uploads.zulipusercontent.net/8f63bc2632a0e41be3f457d86c077e61b4a03e7e/68747470733a2f2f636861742e7a756c69702e6f72672f757365725f617661746172732f322f7265616c6d2f69636f6e2e706e673f76657273696f6e3d34"></a></div>');
+      final images = tester.widgetList<RealmContentNetworkImage>(find.byType(RealmContentNetworkImage));
+      check(images.map((i) => i.src.toString()).toList())
+        .deepEquals([
+          'https://uploads.zulipusercontent.net/f535ba07f95b99a83aa48e44fd62bbb6c6cf6615/68747470733a2f2f636861742e7a756c69702e6f72672f757365725f617661746172732f322f7265616c6d2f69636f6e2e706e673f76657273696f6e3d33',
+          'https://uploads.zulipusercontent.net/8f63bc2632a0e41be3f457d86c077e61b4a03e7e/68747470733a2f2f636861742e7a756c69702e6f72672f757365725f617661746172732f322f7265616c6d2f69636f6e2e706e673f76657273696f6e3d34',
+        ]);
+    });
```

Comment:
I've just sent #511 which should simplify writing this sort of test. If you rebase atop that and then convert this PR's content-parsing tests into the new `ContentExample` form, then for example this test can become:
```suggestion
    testWidgets('multiple images', (tester) async {
      const example = ContentExample.imageMultiple;
      await prepareContent(tester, example.html);
      final expectedImages = (example.expectedNodes[1] as ImageNodeList).images;
      final images = tester.widgetList<RealmContentNetworkImage>(find.byType(RealmContentNetworkImage));
      check(images.map((i) => i.src.toString()))
        .deepEquals(expectedImages.map((n) => n.srcUrl));
    });
```

which I think is a lot easier to digest. The difference, from not duplicating all the HTML of the examples, will be even more when adding those tests of the nested-in-a-list case.

In the code I pasted above, the way `expectedImages` is computed is slightly hacky, and will have to vary somewhat between different test cases depending on the structure of the example content. I'd be fine with a version that's hacky in that way. It'd be neat, but not necessary, to write a bit of logic for this group of tests that walks the tree of `example.expectedNodes` to find the `ImageNodeList` nodes in a generic way.